Hide specific features from nav is dashboard tenant

avatus opened the pull request:
This PR makes an update to our logic to the "backdoor" that would _keep_ things hidden after making updates to show most of the features to promote discoverability. Currently, we only continued to hide them if their license specified it, but this incorrectly ignored dashboard, usage based, and team tenants as well.

mcbattirola commented on web/packages/teleport/src/features.tsx:
```diff
@@ -67,6 +67,23 @@ import { NavTitle, type FeatureFlags, type TeleportFeature } from './types';
+// are usually "always visible"
+export function shouldHideFromNavigation(cfg: {
+  isTeam: boolean;
+  isUsageBasedBilling: boolean;
```
Most subscriptions nowadays should have `isUsageBasedBilling=true` IIRC. The main exceptions are OSS and 14-day trials.

This means that `shouldHideFromNavigation` will return `true` for most customers, is this what we want?

mcbattirola commented on web/packages/teleport/src/features.tsx:
```diff
@@ -67,6 +67,23 @@ import { NavTitle, type FeatureFlags, type TeleportFeature } from './types';
+// However, there are some cases where hiding the feature is explicitly requested. Use this as a backdoor to hide the features that
+// are usually "always visible"
+export function shouldHideFromNavigation(cfg: {
+  isTeam: boolean;
```
I think `isTeam` is hardcoded to `false` now, we use instead `isStripeManaged`.

avatus committed:
Hide specific features from nav is dashboard tenant

This PR makes an update to our logic to the "backdoor" that would keep
things hidden after making updates to show most of the features to
promote discoverability. Currently, we only continued to hide them if
their license specified it, but this incorrectly ignored dashboard,
usage based, and team tenants as well.

avatus commented:
Ok, updated. thanks @mcbattirola
